src/main/java/io/cdap/plugin/google/sink/GoogleDriveSinkClient.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.google.sink;
+
+import com.google.api.client.http.ByteArrayContent;
+import com.google.api.services.drive.model.File;
+import io.cdap.plugin.google.common.FileFromFolder;
+import io.cdap.plugin.google.common.GoogleDriveClient;
+
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import java.util.Collections;
+
+/**
+ * Client for writing data via Google Drive API.
+ */
+public class GoogleDriveSinkClient extends GoogleDriveClient<GoogleDriveSinkConfig> {
+
+  public GoogleDriveSinkClient(GoogleDriveSinkConfig config) {
+    super(config);
+  }
+
+  public void createFile(FileFromFolder fileFromFolder) throws GeneralSecurityException, IOException,
+    InterruptedException {
+    String folderId = config.getDirectoryIdentifier();
+
+    initialize();
+
+    File fileToWrite = new File();
+
+    fileToWrite.setName(fileFromFolder.getFile().getName());
+    fileToWrite.setParents(Collections.singletonList(folderId));
+    ByteArrayContent fileContent = new ByteArrayContent(null, fileFromFolder.getContent());
+    service.files().create(fileToWrite, fileContent).execute();
```

Comment:
Please check if this gives a good readable by end user exceptions. So if this fails when something goes wrong. User can diagnose.

Reply:
Replaced with RuntimeException

Reply:
IOException is allowed for GoogleDriveRecordWriter.write(...), so I this is not need to replace

Reply:
You didn't understand the question. I asked if it gives messages which can be easily understood by user. In case something goes wrong, like disk full, directory deleted or something other.